compiler/src/formatting/format.re

```diff
@@ -1846,46 +1878,96 @@ and print_infix_application =
     let left_needs_parens = left_is_if || left_grouping_required;
     let right_needs_parens = right_is_if || right_grouping_required;
 
-    let wrapped_left =
+    let l1 =
+      print_expression(
+        ~expression_parent=GenericExpression,
+        ~original_source,
+        ~comments,
+        first,
+      );
+
+    let lhs =
       if (left_needs_parens) {
-        if (Doc.willBreak(right_expr)) {
-          Doc.concat([Doc.lparen, left_expr, Doc.rparen]);
-        } else {
-          Doc.concat([Doc.lparen, Doc.indent(left_expr), Doc.rparen]);
-        };
+        Doc.concat([
+          Doc.lparen,
+          if (Doc.willBreak(l1)) {
+            Doc.indent(l1);
+          } else {
+            l1;
+          },
+          Doc.rparen,
+        ]);
       } else {
-        left_expr;
+        print_expression(
+          ~expression_parent,
+          ~original_source,
+          ~comments,
+          first,
+        );
       };
 
-    let wrapped_right =
+    let rhs_expr =
       if (right_needs_parens) {
-        if (Doc.willBreak(right_expr)) {
-          Doc.concat([Doc.lparen, right_expr, Doc.rparen]);
-        } else {
-          Doc.concat([Doc.lparen, Doc.indent(right_expr), Doc.rparen]);
-        };
-      } else {
-        right_expr;
-      };
-
-    let lhs = left_is_leaf ? Doc.group(wrapped_left) : wrapped_left;
-    let rhs = right_is_leaf ? Doc.group(wrapped_right) : wrapped_right;
-
-    let trailing_line_comments =
-      if (line_comment_docs == Doc.nil) {
-        Doc.line;
+        Doc.concat([
+          Doc.lparen,
+          Doc.concat([
+            print_expression(
+              ~expression_parent=GenericExpression,
+              ~original_source,
+              ~comments,
+              second,
+            ),
+          ]),
+          Doc.rparen,
+        ]);
       } else {
-        Doc.concat([Doc.nil, line_comment_docs, Doc.line]);
+        Doc.concat([
+          print_expression(
+            ~expression_parent,
+            ~original_source,
+            ~comments,
+            second,
+          ),
+        ]);
       };
 
     Doc.concat([
-      lhs,
+      if (left_is_leaf) {
+        Doc.group(lhs);
+      } else {
+        lhs;
+      },
       Doc.space,
       Doc.text(function_name),
-      trailing_line_comments,
+      line_comment_docs,
+      if (after_comments_docs == Doc.nil) {
+        Doc.nil;
+      } else {
+        force_break_if_line_comment(~separator=Doc.nil, line_comments);
+      },
       after_comments_docs,
-      force_break_if_line_comment(~separator=Doc.nil, after_comments),
-      rhs,
+      switch (expression_parent) {
+      | ConditionalExpression =>
+        Doc.concat([
+          Doc.line,
+          if (right_is_leaf) {
+            Doc.group(rhs_expr);
+          } else {
+            rhs_expr;
+          },
+        ])
+      | _ =>
```

Comment:
This should not be using a catch-all. That defeats the purpose of turning it into a switch statement